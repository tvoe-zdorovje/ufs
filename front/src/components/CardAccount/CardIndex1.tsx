import React from 'react'
import {
  Row,
  Col,
  Button
} from 'antd'
import renderPair from './RenderPair'

const styles = require('./CardAccount.less')

export interface CardIndex1Props {
  items: Array<any>,
  addSidePanel: Function,
}

export default function CardIndex1({ items, addSidePanel }: CardIndex1Props) {
  const listItems = items.map((item, i) => {
    const pairs = [['Номер документа', item.docId],
      ['Вид', item.docType],
      ['Получатель', item.recipientShortName],
      ['Сумма', item.amount],
      ['Дата', item.comeInDate]];
    return (
      <Row className={styles.separatedRow} key={i}>
        <Col span={23}>
          {pairs.map((pair, j) => renderPair(pair[0], pair[1], j))}
        </Col>
        <Col span={1}>
          <Button
            className={styles.largeIcon}
            onClick={() => addSidePanel(pairs, `ПП ${item.docId}`)}
            icon="menu-unfold"
          />
        </Col>
      </Row>
    )
  })

  return (
    <div>
      {listItems}
    </div>
  )
}
