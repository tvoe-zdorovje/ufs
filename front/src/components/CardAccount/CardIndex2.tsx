import React from 'react'
import {
  Row,
  Col,
  Button
} from 'antd'
import renderPair from './RenderPair'

const styles = require('./CardAccount.less')

export interface CardIndex2Props {
  items: any[],
  addSidePanel: Function,
}

export default function CardIndex2({ items, addSidePanel }: CardIndex2Props) {
  const listItems = items.map((item, i) => {
    const pairs = [['Номер документа', item.docId],
      ['Очередь оплаты', item.priority],
      ['Сумма по документу', item.totalAmount],
      ['Оплачено', item.partAmount],
      ['Получатель', item.recipientShortName]]
    return (
      <Row className={styles.separatedRow} key={i}>
        <Col span={23}>
          {pairs.filter((pair) => pair[1]).map((pair, j) => renderPair(pair[0], pair[1], j))}
        </Col>
        <Col span={1}>
          <Button
            className={styles.largeIcon}
            onClick={() => addSidePanel(pairs, `ПП ${item.docId}`)}
            icon="menu-unfold"
          />
        </Col>
      </Row>
    )
  })

  return (
    <div>
      {listItems}
    </div>
  )
}
