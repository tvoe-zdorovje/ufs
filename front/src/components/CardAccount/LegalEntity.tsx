import React from 'react'
import {
  Row,
  Col
} from 'antd'
import renderPair from './RenderPair'
import { LegalEntity } from 'interfaces/dto/Account.interface'

export interface LegalEntityProps {
  legalEntity: LegalEntity
}

export default function LegalEntity({ legalEntity }: LegalEntityProps) {
  const pairs = [['Наименование', legalEntity.fullName],
    ['Сокращенное наименование', legalEntity.shortName],
    ['ИНН', legalEntity.inn],
    ['ОГРН', legalEntity.ogrn],
    ['КПП', legalEntity.kpp],
    ['Юридический адрес', legalEntity.legalAddress],
    ['Фактический адрес', legalEntity.factAddress]]

  return (
    <Row>
      <Col span={23}>
        {pairs.map((item, i) => renderPair(item[0], item[1], i))}
      </Col>
    </Row>
  )
}
