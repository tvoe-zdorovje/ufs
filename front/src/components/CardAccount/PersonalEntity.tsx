import React from 'react'
import {
  Row,
  Col,
  Button
} from 'antd'
import renderPair from './RenderPair'
import {
  AccountResidues,
  PersonalAccount,
  Seizure
} from 'interfaces/dto/Account.interface'

const styles = require('./CardAccount.less')

const getDetails = (accountResidues: AccountResidues) => ([['Текущий остаток', accountResidues.currentBalance],
    ['Неснижаемый остаток', accountResidues.fixedBalance],
    ['Планируемый остаток', accountResidues.expectedBalance]]
)

const getSeizures = (seizure) => (
  [['Сумма ареста', seizure.amount],
    ['Основание ареста ', seizure.reason],
    ['Срок ареста', `С ${seizure.fromDate} По ${seizure.toDate}`],
    ['ЮЛ/орган-инициатор', seizure.initiatorShortName],
  ])

const renderStatus = (info, accountResidues, addSidePanel, seizures) => (
  <Row className={(info.status === 'Active') ? styles.green : styles.red}>
    <Col span={8}>
      {(info.status === 'Active') ? 'Активный' : 'Неактивный'}
    </Col>
    <Col span={10} className={styles.rightAligned}>
      <Button
        className={`${styles.green} ${styles.noBorder}`}
        onClick={() => addSidePanel(getDetails(accountResidues), 'Доп. информация по счету')}
      >
        Подробнее
      </Button>
    </Col>
    <Col span={6}>
      {(seizures && seizures.length > 0) ? (
        <Button
          className={`${styles.red} ${styles.noBorder}`}
          onClick={() => addSidePanel(getSeizures(seizures[0]), 'Информация об арестах')}
        >
          Арест суммы
        </Button>) : null}
    </Col>
  </Row>
)

export interface PerosnalEntityProps {
  info: PersonalAccount,
  accountResidues: AccountResidues,
  addSidePanel: Function,
  seizures: Array<Seizure>,
}

export default function PerosnalEntity({ info, accountResidues, addSidePanel, seizures }: PerosnalEntityProps) {
  const pairs = [['Номер счета', info.id],
    ['Вид', 'Расчетный счет'],
    ['Номер договора', info.agreement.id],
    ['Текущий остаток', accountResidues.currentBalance],
    ['Валюта', info.currencyType],
    ['Дата последней операции', info.lastTransactionDate],
    ['Статус', renderStatus(info, accountResidues, addSidePanel, seizures)]]

  return (
    <Row>
      <Col span={23}>
        {pairs.map((item, i) => renderPair(item[0], item[1], i))}
      </Col>
    </Row>
  )
}
