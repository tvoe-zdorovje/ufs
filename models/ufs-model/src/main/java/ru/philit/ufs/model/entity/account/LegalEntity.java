package ru.philit.ufs.model.entity.account;

import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.Setter;
import lombok.ToString;
import ru.philit.ufs.model.entity.common.ExternalEntity;

/**
 * Сущность Юридического лица.
 */
@EqualsAndHashCode(of = {"id"}, callSuper = false)
@ToString
@Getter
@Setter
public class LegalEntity extends ExternalEntity {

  private String id;
  private String shortName;
  private String fullName;
  private String inn;
  private String ogrn;
  private String kpp;
  private String legalAddress;
  private String factAddress;

}
