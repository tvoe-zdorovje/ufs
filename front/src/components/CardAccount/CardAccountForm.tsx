import React from 'react'
import {
  Button,
  Col,
  Collapse,
  Icon,
  Row
} from 'antd'
import LegalEntity from './LegalEntity'
import PersonalEntity from './PersonalEntity'
import CardIndex1 from './CardIndex1'
import CardIndex2 from './CardIndex2'
import CloseableCollapse, { PairsItem } from './CloseableCollapse'
import { Account } from 'interfaces/dto/Account.interface'

const styles = require('./CardAccount.less')

const Panel = Collapse.Panel

export interface CardAccountFormProps {
  getCardAccount: Function,
  addSidePanel: Function,
  onComplete: Function,
  removeSidePanel: Function,
  loaded?: boolean,
  info?: Account,
  sidePanel: Array<PairsItem>,
  cardInfo: any,
}

export default class CardAccountForm extends React.Component<CardAccountFormProps, any> {
  componentDidMount() {
    if (this.props.loaded) {
      return
    }
    this.props.getCardAccount({ cardNumber: this.props.cardInfo.number })
  }

  render() {
    const { info, cardInfo, addSidePanel, onComplete, removeSidePanel } = this.props
    return (
      <div>
        <Row>
          <Col span={10}>
            {`${cardInfo.ownerFirstName} ${cardInfo.ownerLastName}`}
          </Col>
          <Col span={8} className={styles.rightAligned}>
            {`VISA ${cardInfo.number}`}
          </Col>
        </Row>
        <Row>
          <Col span={18}>
            <Collapse className={styles.accountCard} defaultActiveKey={['2']}>
              <Panel header="Общая информаци о юридическом лице" key="1">
                {info.legalEntity ?
                  <LegalEntity legalEntity={info.legalEntity} /> :
                  null
                }
              </Panel>
              <Panel header="Общая информация по счету" key="2">
                {info.account ? <PersonalEntity
                  addSidePanel={addSidePanel}
                  info={info.account}
                  seizures={info.seizures}
                  accountResidues={info.accountResidues}
                />
                  :
                  null
                }
              </Panel>
              <Panel header={`Картотека №1 - ${info.cardIndexes1.length} ожидают утверждения`} key="3">
                <CardIndex1 addSidePanel={addSidePanel} items={info.cardIndexes1} />
              </Panel>
              <Panel header={`Картотека №2 - ${info.cardIndexes2.length} в очереди`} key="4">
                <CardIndex2 addSidePanel={addSidePanel} items={info.cardIndexes2} />
              </Panel>
            </Collapse>
          </Col>
          <Col span={6}>
            <CloseableCollapse
              sidePanel={this.props.sidePanel}
              removePanel={removeSidePanel}
            />
          </Col>
        </Row>
        <Row>
          <Col span={18} className={styles.rightAligned}>
            <Button type="primary" disabled={!info.account} onClick={() => onComplete()}>
              Далее<Icon type="right" />
            </Button>
          </Col>
          <Col span={6} />
        </Row>
      </div>
    )
  }
}
