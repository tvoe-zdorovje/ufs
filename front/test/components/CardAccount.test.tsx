import React from 'react'
import {
  mount,
  render,
  shallow
} from 'enzyme'
import toJson from 'enzyme-to-json'

import { Collapse } from 'antd'

import {
  Account,
} from 'interfaces/dto/Account.interface'
import CardAccountForm, { CardAccountFormProps } from 'components/CardAccount/CardAccountForm'
import PersonalEntity from 'components/CardAccount/PersonalEntity'
import LegalEntity from 'components/CardAccount/LegalEntity'
import CardIndex1 from 'components/CardAccount/CardIndex1'
import CardIndex2 from 'components/CardAccount/CardIndex2'

const mockFn = () => {
}

describe('Компонент CardAccountForm', () => {

  const account: Account = {
    legalEntity: {
      id: '', fullName: 'OOO ДНК', shortName: 'OOO ДНК', inn: '01221412', ogrn: '102312412', kpp: '121412412',
      legalAddress: 'Ярославль', factAddress: 'Ярославль'
    },
    account: {
      id: '4782 9871 9873 897009',
      typeId: 'Расчетный счет',
      agreement: {
        id: '124567',
        openDate: '',
      },
      status: '1',
      lastTransactionDate: '18.06.2001',
      currencyType: 'RUB',
      active: true,
      tbCode: '',
      gosbCode: '',
      vspCode: '',
      subbranchCode: '',
      changeStatusDescription: '',
      clientTypeFk: true,
      seizured: true,
    },
    accountResidues: {
      currentBalance: '1',
      expectedBalance: '2',
      fixedBalance: '3',
    },
    cardIndexes1: [{
      docId: '98710245', docType: 'Платежное поручение', recipientShortName: 'Газпром', amount: '20000',
      comeInDate: '01.05.2017'
    }, {
      docId: '98710246', docType: 'Платежное поручение', recipientShortName: 'Газпром', amount: '20000',
      comeInDate: '01.05.2017'
    }],
    cardIndexes2: [{
      docId: '1',
      docType: '2',
      recipientShortName: '3',
      amount: '4',
      totalAmount: '5',
      paidPartly: false,
      comeInDate: '01.05.2017',
      priority: '1',
      paids: [{ date: '1', amount: '2' }]
    }]
  }

  const props: CardAccountFormProps = {
    loaded: true,
    info: account,
    sidePanel: [{ pairs: ['0', '1'] }],
    cardInfo: {},
    getCardAccount: mockFn,
    addSidePanel: (x, y) => {
    },
    removeSidePanel: mockFn,
    onComplete: mockFn,
  }

  const renderWrapper = render(<CardAccountForm {...props} />)
  const shallowWrapper = shallow(<CardAccountForm {...props} />)

  it('отображает контент - компонент <Collapse />', () => {
    expect(shallowWrapper.find(Collapse)).toHaveLength(1)
  })

  it('отображает контент - компонент <LegalEntity />', () => {
    expect(shallowWrapper.find(LegalEntity)).toHaveLength(1)
  })

  it('отображает контент - компонент <PersonalEntity />', () => {
    expect(shallowWrapper.find(PersonalEntity)).toHaveLength(1)
  })

  it('отображает контент - компонент <CardIndex1 />', () => {
    expect(shallowWrapper.find('CardIndex1')).toHaveLength(1)
  })

  it('отображает контент - компонент <CardIndex2 />', () => {
    expect(shallowWrapper.find('CardIndex2')).toHaveLength(1)
  })

  it('не отображает юридическую информация о лице если ее нет', () => {
    shallowWrapper.setProps({ info: { ...props.info, legalEntity: undefined } })
    shallowWrapper.update()
    expect(shallowWrapper.find(LegalEntity) || []).toHaveLength(0)
  })

  it('не отображает общую информацию по счету если ее нет', () => {
    shallowWrapper.setProps({ info: { ...props.info, account: undefined } })
    shallowWrapper.update()
    expect(shallowWrapper.find(PersonalEntity) || []).toHaveLength(0)
  })

  it('запрашивает getCardAccount если не загружено', () => {
    const spy = jest.fn()
    const getCardAccountSpy = { ...props, getCardAccount: spy, loaded: false }
    mount(<CardAccountForm { ...getCardAccountSpy} />)
    expect(spy).toBeCalled()
  })

  it('совпадает со своим снимком', () => {
    expect(toJson(renderWrapper)).toMatchSnapshot()
  })
})
