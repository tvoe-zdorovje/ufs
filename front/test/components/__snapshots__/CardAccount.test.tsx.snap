// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Компонент CardAccountForm совпадает со своим снимком 1`] = `
<div>
  <div
    class="ant-row"
  >
    <div
      class="ant-col-10"
    >
      undefined undefined
    </div>
    <div
      class="ant-col-8 rightAligned"
    >
      VISA undefined
    </div>
  </div>
  <div
    class="ant-row"
  >
    <div
      class="ant-col-18"
    >
      <div
        class="ant-collapse accountCard"
      >
        <div
          class="ant-collapse-item"
        >
          <div
            aria-expanded="false"
            class="ant-collapse-header"
            role="tab"
          >
            <i
              class="arrow"
            />
            Общая информаци о юридическом лице
          </div>
        </div>
        <div
          class="ant-collapse-item ant-collapse-item-active"
        >
          <div
            aria-expanded="true"
            class="ant-collapse-header"
            role="tab"
          >
            <i
              class="arrow"
            />
            Общая информация по счету
          </div>
          <div
            class="ant-collapse-content ant-collapse-content-active"
            role="tabpanel"
          >
            <div
              class="ant-collapse-content-box"
            >
              <div
                class="ant-row"
              >
                <div
                  class="ant-col-23"
                >
                  <div
                    class="ant-row pairRow"
                  >
                    <div
                      class="ant-col-8 rightAligned bold"
                    >
                      Номер счета
                    </div>
                    <div
                      class="ant-col-16"
                    >
                      4782 9871 9873 897009
                    </div>
                  </div>
                  <div
                    class="ant-row pairRow"
                  >
                    <div
                      class="ant-col-8 rightAligned bold"
                    >
                      Вид
                    </div>
                    <div
                      class="ant-col-16"
                    >
                      Расчетный счет
                    </div>
                  </div>
                  <div
                    class="ant-row pairRow"
                  >
                    <div
                      class="ant-col-8 rightAligned bold"
                    >
                      Номер договора
                    </div>
                    <div
                      class="ant-col-16"
                    >
                      124567
                    </div>
                  </div>
                  <div
                    class="ant-row pairRow"
                  >
                    <div
                      class="ant-col-8 rightAligned bold"
                    >
                      Текущий остаток
                    </div>
                    <div
                      class="ant-col-16"
                    >
                      1
                    </div>
                  </div>
                  <div
                    class="ant-row pairRow"
                  >
                    <div
                      class="ant-col-8 rightAligned bold"
                    >
                      Валюта
                    </div>
                    <div
                      class="ant-col-16"
                    >
                      RUB
                    </div>
                  </div>
                  <div
                    class="ant-row pairRow"
                  >
                    <div
                      class="ant-col-8 rightAligned bold"
                    >
                      Дата последней операции
                    </div>
                    <div
                      class="ant-col-16"
                    >
                      18.06.2001
                    </div>
                  </div>
                  <div
                    class="ant-row pairRow"
                  >
                    <div
                      class="ant-col-8 rightAligned bold"
                    >
                      Статус
                    </div>
                    <div
                      class="ant-col-16"
                    >
                      <div
                        class="ant-row red"
                      >
                        <div
                          class="ant-col-8"
                        >
                          Неактивный
                        </div>
                        <div
                          class="ant-col-10 rightAligned"
                        >
                          <button
                            class="ant-btn green noBorder"
                            type="button"
                          >
                            <span>
                              Подробнее
                            </span>
                          </button>
                        </div>
                        <div
                          class="ant-col-6"
                        />
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="ant-collapse-item"
        >
          <div
            aria-expanded="false"
            class="ant-collapse-header"
            role="tab"
          >
            <i
              class="arrow"
            />
            Картотека №1 - 2 ожидают утверждения
          </div>
        </div>
        <div
          class="ant-collapse-item"
        >
          <div
            aria-expanded="false"
            class="ant-collapse-header"
            role="tab"
          >
            <i
              class="arrow"
            />
            Картотека №2 - 1 в очереди
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-col-6"
    >
      <div
        class="ant-collapse accountCard"
      >
        <div
          class="ant-collapse-item ant-collapse-item-active"
        >
          <div
            aria-expanded="true"
            class="ant-collapse-header"
            role="tab"
          >
            <i
              class="arrow"
            />
            <div
              class="ant-row"
            >
              <div
                class="ant-col-21"
              />
              <div
                class="ant-col-3"
              >
                <button
                  class="ant-btn ant-btn-circle"
                  type="button"
                >
                  <span>
                    X
                  </span>
                </button>
              </div>
            </div>
          </div>
          <div
            class="ant-collapse-content ant-collapse-content-active"
            role="tabpanel"
          >
            <div
              class="ant-collapse-content-box"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="ant-row"
  >
    <div
      class="ant-col-18 rightAligned"
    >
      <button
        class="ant-btn ant-btn-primary"
        type="button"
      >
        <span>
          Далее
        </span>
        <i
          class="anticon anticon-right"
        />
      </button>
    </div>
    <div
      class="ant-col-6"
    />
  </div>
</div>
`;
